app/controllers/users_controller.rb

```diff
@@ -38,13 +38,17 @@ def set_password
   end
 
   def save_password
-    @user = User.find_by_token(params[:token])
-    if @user && @user.update_attributes(params[:user].slice(:password, :password_confirmation, :phone))
-      @user.reload
-      login(@user)
-      redirect_to dashboard_url
+    if not UsersHelper.secure_password?(params[:user].slice(:password))
```

Comment:
`if not else` is generally discouraged. You can flip the inner code blocks and remove the `not`.

There is an argument to be made for leaving it as is (it has our early exit at the start and then the meat of the method after). In that case, you can change `if not` to `unless`, which is more idiomatic.